From 01291778789ed8b0279834fd89cbc67db2f6820c Mon Sep 17 00:00:00 2001
From: Nathan Froyd <froydnj@mozilla.com>
Date: Thu, 19 Sep 2019 17:33:26 +0000
Subject: [PATCH 03/35] Bug 1571559 - add configuration support for
 --with-wasi-sysroot; r=nalexander

As part of compiling C/C++ to wasm, we're going to need a wasm-specific
sysroot to be provided.  This option enables us to specify that sysroot
during configure.

Differential Revision: https://phabricator.services.mozilla.com/D46314

--HG--
extra : moz-landing-system : lando

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 709667f115f7..e2ba6640e8af 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1792,6 +1792,27 @@ def wasm_sandboxing_config_defines():
 
 wasm_sandboxing_config_defines()
 
+option('--with-wasi-sysroot',
+       nargs=1,
+       help='Path to wasi sysroot for wasm sandboxing',
+       when=requires_wasm_sandboxing)
+
+@depends('--with-wasi-sysroot', when=requires_wasm_sandboxing)
+@imports('os')
+def wasi_sysroot(wasi_sysroot):
+    if not wasi_sysroot:
+        die('--with-wasi-sysroot must be specified when using wasm sandboxing')
+
+    wasi_sysroot = wasi_sysroot[0]
+    if not os.path.isdir(wasi_sysroot):
+        die('Argument to --with-wasi-sysroot must be a directory')
+    if not os.path.isabs(wasi_sysroot):
+        die('Argument to --with-wasi-sysroot must be an absolute path')
+
+    return wasi_sysroot
+
+set_config('WASI_SYSROOT', wasi_sysroot)
+
 
 # new XULStore implementation
 # ==============================================================
-- 
2.25.1

